src/test/java/net/floodlightcontroller/statistics/GetCurrentPortSpeedTest.java

```diff
@@ -0,0 +1,145 @@
+    private static final OFFactory factory12 = OFFactories.getFactory(OFVersion.OF_12);
+    private static final OFFactory factory13 = OFFactories.getFactory(OFVersion.OF_13);
+    private static final OFFactory factory14 = OFFactories.getFactory(OFVersion.OF_14);
+    private static final OFFactory factory15 = OFFactories.getFactory(OFVersion.OF_15);
+    private static OFFactory inputFactory;
+    private static Long expectedSpeed;
+
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+
+        cntx = new FloodlightContext();
+        fmc = new FloodlightModuleContext();
+
+        // Module loader setup
+        threadpool = new MockThreadPoolService();
+        switchService = getMockSwitchService();
+        statsCollector = new StatisticsCollector();
+
+        fmc.addService(IThreadPoolService.class, threadpool);
+        fmc.addService(IOFSwitchService.class, switchService);
+
+        threadpool.init(fmc);
+        statsCollector.init(fmc);
+        threadpool.startUp(fmc);
+
+    }
+
+    /**
+     * This constructor will be called for each row of test data collection
+     * @param inputFactory
+     * @param expectedSpeed
+     */
+    public GetCurrentPortSpeedTest(OFFactory inputFactory, Long expectedSpeed) {
+        this.inputFactory = inputFactory;
+        this.expectedSpeed = expectedSpeed;
+    }
+
+    /**
+     * A Collection of Junit Test with various "inputFactory" and "expectedSpeed"
+     * @return
+     */
+    @Parameterized.Parameters
+    public static Iterable<Object[]> testData() {
+        return Arrays.asList(new Object[][] {
+            { factory11, 100L },
+            { factory12, 100L },
+            { factory13, 100L },
+            { factory14, 100L },
+            { factory15, 100L },
+        });
+    }
+
+    /**
+     * Test getSpeed() method works with different Openflow versions
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testGetCurrentPortSpeed() throws Exception {
+        IOFSwitch sw = getSwitchByOFVersion(inputFactory);
+        NodePortTuple npt = new NodePortTuple(DatapathId.of(1), OFPort.of(1));
+        Map<DatapathId, IOFSwitch> switchMap = new HashMap<>();
+        switchMap.put(sw.getId(), sw);
+        getMockSwitchService().setSwitches(switchMap);
+
+        StatisticsCollector.PortStatsCollector statsSpeedCollector = statsCollector.new PortStatsCollector();
+        Long speed = statsSpeedCollector.getSpeed(npt);
+
+        assertEquals(speed, expectedSpeed);
+
+    }
+
+    private IOFSwitch getSwitchByOFVersion(OFFactory inputFactory) {
+        IOFSwitch sw = EasyMock.createMock(IOFSwitch.class);
+        reset(sw);
+        expect(sw.getId()).andReturn(DatapathId.of(1L)).anyTimes();
+        expect(sw.getOFFactory()).andReturn(inputFactory).anyTimes();
+
+        OFVersion factoryVersion = inputFactory.getVersion();
+        switch (factoryVersion){
+            case OF_11:
```

Comment:
Going with the `OFVersion` comment above, if you just passed in the version instead of the factory, you wouldn't need to extract the version out of the factory here